packages/server/graphql/public/mutations/stopRecurrence.ts

```diff
@@ -0,0 +1,57 @@
+import getRethink from '../../../database/rethinkDriver'
+import updateMeetingSeries from '../../../postgres/queries/updateMeetingSeries'
+import {getUserId, isTeamMember} from '../../../utils/authorization'
+import standardError from '../../../utils/standardError'
+import {MutationResolvers} from '../resolverTypes'
+
+const stopRecurrence: MutationResolvers['stopRecurrence'] = async (
+  _source,
+  {meetingId},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const r = await getRethink()
+  const viewerId = getUserId(authToken)
+  const now = new Date()
+
+  // VALIDATION
+  const meeting = await dataLoader.get('newMeetings').load(meetingId)
+  if (!meeting) {
+    return standardError(new Error('Meeting not found'), {userId: viewerId})
+  }
+
+  const {teamId, meetingType} = meeting
+
+  if (!isTeamMember(authToken, teamId)) {
+    return standardError(new Error('Team not found'), {userId: viewerId})
+  }
+
+  if (meetingType !== 'teamPrompt') {
+    return standardError(new Error('Meeting is not a team prompt meeting'), {userId: viewerId})
+  }
+
+  if (!meeting.meetingSeriesId) {
+    return standardError(new Error('Meeting does not have meeting series associated!'), {
+      userId: viewerId
+    })
+  }
+
+  const meetingSeries = await dataLoader.get('meetingSeries').loadNonNull(meeting.meetingSeriesId)
+  await updateMeetingSeries({cancelledAt: now}, meetingSeries.id)
+
+  dataLoader.get('meetingSeries').clear(meeting.meetingSeriesId)
+
+  await r
+    .table('NewMeeting')
+    .filter({meetingSeriesId: meetingSeries.id})
+    .filter({endedAt: null}, {default: true})
+    .update({
+      scheduledEndTime: null
+    })
+    .run()
```

Comment:
Is there any better way of writing this query? should we worry about performance here?

Reply:
to be fair, this mutation should just take `meetingSeriesId` as a param, not a single meeting. How can we then make sure to remove `scheduledEndTime` from all the existing open meetings not to kill our server? 🤔

Reply:
should we put `meetingSeriesId` and  `scheduledEndTime` on a `Meeting` type instead of `MeetingTeamPrompt` so we can use `getAll(meetingSeriesId, {index: 'meetingSeriesId})`?

Reply:
@mattkrick I've added a `meetingSeriesId` index in the last commit, do you mind looking one more time? thank you!

Reply:
+1 ah! sorry i didn't scroll up to see this thread.
if we want to apply this to all meetings within a meetingSeries I'd probably change the argument to `meetingSeriesId` instead of `meetingId` to be a little more expressive, but the extra meeting lookup is inconsequential